app/models/user.rb

```diff
@@ -328,6 +328,10 @@ def increment_subjects_count_cache
     Rails.cache.increment(subjects_count_cache_key)
   end
 
+  def favorite_collections_for_project(project)
+    collections.where(favorite: true, projects: [project])
```

Comment:
This seems like it has feature envy with collections. I think it would be great to extract all this to `FavoritesFinder.new(api_user, project).ids` (lib / model concern?) that deals with this logic.